internal/bufferedread/buffered_reader.go

```diff
@@ -98,6 +98,44 @@ func NewBufferedReader(object *gcs.MinObject, bucket gcs.Bucket, config *Buffere
 	return reader, nil
 }
 
+// ScheduleNextBlock schedules the next block for prefetch.
+func (p *BufferedReader) ScheduleNextBlock(urgent bool) error {
+	b, err := p.blockPool.Get()
+	if err != nil {
```

Comment:
We can return custom error in case block not available so that ReadAt() method can take decision either to fail or fallback.